messaging/connectors/kafka/src/main/java/io/helidon/messaging/connectors/kafka/KafkaSubscriber.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2020 Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.messaging.connectors.kafka;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import io.helidon.config.Config;
+
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.Producer;
+import org.eclipse.microprofile.reactive.messaging.Message;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+/**
+ * Reactive streams subscriber implementation.
+ *
+ * @param <T> kafka record value type
+ */
+class KafkaSubscriber<T> implements Subscriber<Message<T>> {
+
+    private static final Logger LOGGER = Logger.getLogger(KafkaSubscriber.class.getName());
+    private static final String BACKPRESSURE_SIZE_KEY = "backpressure.size";
+    private static final long BACKPRESSURE_SIZE_DEFAULT = 5;
+    private final long backpressure;
+    private final AtomicLong backpressureCounter = new AtomicLong();
+    private final BasicKafkaProducer<?, T> producer;
+    private Subscription subscription;
+
+    private KafkaSubscriber(BasicKafkaProducer<?, T> producer, long backpressure){
+        this.backpressure = backpressure;
+        this.producer = producer;
+    }
+
+    @Override
+    public void onSubscribe(Subscription subscription) {
+        if (this.subscription == null) {
+            this.subscription = subscription;
+            this.subscription.request(backpressure);
```

Comment:
Backpressure is not driven by anything here, use `.request(Long.MAX)` instead pls. no need for request counting

Reply:
Backpressure has its place 
 - when reading messages from Kafka and delivering them to consumers, we should not deliver more than requested
 - when writing message to Kafka and reading them from producers, we should not request more than we can send at that time

In both cases, requesting `Long.MAX` may result in memory issues, as the messages need to be buffered, or in thread issues, as you would block threads.

Reply:
No doubt about that, without taking result of `producer.produceAsync(message.getPayload());` in to the account its just unbounded stream with more bureaucracy

Reply:
Jorge already came with nice idea of combining callbacks